internal/services/compute/managed_disk_resource.go

```diff
@@ -715,14 +715,16 @@ func resourceManagedDiskUpdate(d *pluginsdk.ResourceData, meta interface{}) erro
 				state = strings.TrimPrefix(state, "powerstate/")
 				switch strings.ToLower(state) {
 				case "deallocated":
-				case "deallocating":
-					shouldTurnBackOn = false
+					// VM already deallocated, no shutdown and deallocation needed anymore
 					shouldShutDown = false
 					shouldDeallocate = false
-				case "stopping":
```

Comment:
setting `shouldTurnBackOn` to `false` is not needed in case of `stopping` as `virtualMachineShouldBeStarted` already returns false when state is not `running`